config_gnome2.c

```diff
@@ -34,23 +34,29 @@ typedef struct g_proxy_config_gnome2_s {
 
 g_proxy_config_gnome2_s g_proxy_config_gnome2;
 
-bool proxy_config_gnome2_get_auto_discover(void) {
-    char *mode = NULL;
-    bool auto_discover = false;
+static bool proxy_config_gnome2_is_mode(const char *mode) {
+    bool equal = false;
 
-    mode =
+    char *compare_mode =
         g_proxy_config_gnome2.gconf_engine_get_string(g_proxy_config_gnome2.gconf_default, "/system/proxy/mode", NULL);
-    if (mode) {
-        auto_discover = strcmp(mode, "auto");
-        g_proxy_config_gnome2.g_free(mode);
+    if (compare_mode) {
+        equal = strcmp(compare_mode, mode);
```

Comment:
Isn't an error here? Shouldn't be `equal = strcmp(compare_mode, mode) == 0;`?

Reply:
Fixed. Nice catch.